HoloPipelines/jobs/jobs_io.py

```diff
@@ -10,55 +10,106 @@
     return [
         f"{job_directory_path}/{subdirectory_name}"
-        for subdirectory_name in job_subdirectories
+        for subdirectory_name in subdirectories_per_job
     ]
 
 
 def get_temp_file_path_for_job(job_id: str, file_name: str):
-    return f"{job_directories_root}/{job_id}/temp/{file_name}"
+    return f"{jobs_root}/{job_id}/temp/{file_name}"
 
 
 def get_result_file_path_for_job(job_id: str):
-    return f"{job_directories_root}/{job_id}/output/out.glb"
+    return f"{jobs_root}/{job_id}/output/out.glb"
 
 
 def get_input_directory_path_for_job(job_id: str):
-    return f"{job_directories_root}/{job_id}/input"
+    return f"{jobs_root}/{job_id}/input"
+
+
+def init_create_job_state_root_directories() -> None:
+    logging.info("Creating job state root directories")
+    os.makedirs(jobs_root, exist_ok=True)
+    os.makedirs(finished_jobs_root, exist_ok=True)
 
 
 def create_directory_for_job(job_id: str):
     logging.info(f"Creating directory for job '{job_id}'")
     job_directory_path = get_directory_path_for_job(job_id)
-    os.makedirs(job_directory_path, exist_ok=True)
+    os.mkdir(job_directory_path)
 
     for subdirectory_path in get_subdirectories_paths_for_job(job_id):
         if not os.path.isdir(subdirectory_path):
             os.mkdir(subdirectory_path)
 
 
-def remove_directory_for_job(job_id: str, keep_log_file: bool = True):
-    logging.info(f"Removing files for job '{job_id} (keep_log_file={keep_log_file})'")
-    job_directory_path = get_directory_path_for_job(job_id)
-    if keep_log_file:
-        for subdirectory_path in get_subdirectories_paths_for_job(job_id):
-            shutil.rmtree(subdirectory_path)
-    else:
-        shutil.rmtree(job_directory_path)
+def move_job_to_finished_jobs_directory(job_id: str) -> None:
+    old_path = get_directory_path_for_job(job_id)
+    new_path = old_path.replace(jobs_root, finished_jobs_root)
+    logging.info(f"Moving '{old_path}' to '{new_path}'")
+    shutil.move(old_path, new_path)
+
+
+def remove_temporary_data_for_job(job_id: str) -> None:
+    """
+    Removes all transient job data except for the log file.
+    """
+    logging.info(f"Removing temporary files for job '{job_id}")
+    # input, output, temp
+    for subdirectory_path in get_subdirectories_paths_for_job(job_id):
+        shutil.rmtree(subdirectory_path)
+    # job.state
+    os.remove(get_state_file_path_for_job(job_id))
+
+
+def remove_log_file_for_job(job_id: str) -> None:
+    logging.info(f"Removing log file for job '{job_id}")
+    os.remove(get_log_file_path_for_job(job_id))
+
+
+def get_all_job_subdirectory_names() -> List[str]:
+    return [dir.name for dir in os.scandir(jobs_root) if dir.is_dir()]
+
+
+def read_state_file_for_job(jod_id: str) -> Tuple[str, float]:
+    """
+    Reads state file and returns state and time in seconds since last modification.
+    """
+    state_file_path = get_state_file_path_for_job(jod_id)
+    with open(state_file_path, "r") as state_file:
```

Comment:
There needs to be error handling here, as querying a non-existent job id will throw an exception

Reply:
Agree, did it in the other branch